assets/src/editor/events/ticket-assignments-manager/ticket-assignments-handler.js

```diff
@@ -1,76 +1,56 @@
 /**
  * External imports
  */
-import { find, findIndex, isArray, isEmpty, isUndefined, omitBy } from 'lodash';
-import { isModelEntityOfModel } from '@eventespresso/validators';
-import { normalizeEntityId } from '@eventespresso/helpers';
+import { isArray, isEmpty, isUndefined, omitBy } from 'lodash';
 
 const noIndex = -1;
 
 /**
  * @function
  * @param {Object} assigned
- * @param {Object} date
- * @param {Object} ticket
+ * @param {number} dateId
+ * @param {number} ticketId
  * @param {boolean} returnIndex
  * @return {number|boolean} index for date id in assigned array or
  *                        boolean if returnIndex is false
  */
-export const isAssigned = ( assigned, date, ticket, returnIndex = false ) => {
+export const isAssigned = ( assigned, dateId, ticketId, returnIndex = false ) => {
 	let index = noIndex;
-	if ( isArray( assigned[ date.id ] ) ) {
-		index = findIndex( assigned[ date.id ], { id: ticket.id } );
+	if ( isArray( assigned[ dateId ] ) ) {
+		index = assigned[ dateId ].indexOf( ticketId );
```

Comment:
The main change in the handler file is to only deal with entity ids instead of full entities.  The primary reason for this is that full entities are _not_ needed and it makes it unnecessarily more complicated when searching for values.  The trigger for this is that `prevState` is being mutated in `assignTicketEntity` and `removeTicketEntity`.  State should never be mutated directly (note that `TicketAssignmentsManagerModal` is passing in a deep cloned `prevState` to the handler).